ports/stm32/stm_dma.c

```diff
@@ -357,6 +357,24 @@ int stm_dma_init(DMA_HandleTypeDef *dma_descr, void *dma_channel, uint32_t reque
         if (HAL_DMA_Init(dma_descr) != HAL_OK) {
             return -1;
         }
+
+        #if defined(STM32N6)
```

Comment:
This should be done in `stm_dma_init` for regular and circular modes, if it's okay to configure the security stuff before the rest of the code in `stm_dma_ll_init`, otherwise it should be a new function: `stm_dma_sec_config`

Reply:
Added a new function `stm_dma_sec_config`.

Reply:
Can we configure security before `stm_dma_ll_init` gets called? If so we can just configure it for N6 in this function unconditionally, before `HAL_DMA_Init`.

Reply:
Updated with the requested changes. However, uh... it's kinda dependent on the HAL not changing on us. The functions can be called right now before init... but, that might not be the case in the future. The HAL doesn't check the ready state flag for setting the security config. They could always add that, though.

Reply:
I think the only restriction on these registers is that the channel must be disabled to update them, so it doesn't matter where it's called. Also, now we don't need a separate function.